pinot-connectors/pinot-spark-2-connector/pom.xml

```diff
@@ -35,6 +35,45 @@
   </properties>
 
   <profiles>
+    <profile>
+      <id>build-shaded-jar</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-shade-plugin</artifactId>
+            <executions>
+              <execution>
+                <phase>package</phase>
+                <goals>
+                  <goal>shade</goal>
+                </goals>
+                <configuration>
+                  <relocations>
+                    <relocation>
+                      <pattern>com</pattern>
+                      <shadedPattern>${shadeBase}.com</shadedPattern>
+                      <includes>
+                        <include>com.google.protobuf.**</include>
+                        <include>com.google.common.**</include>
+                      </includes>
+                    </relocation>
+                  </relocations>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>pinot-fastdev</id>
```

Comment:
I actually feel we can skip spark connectors modules for fastdev, since the usage of them usually requires the shaded jar.

Reply:
What we can do then is to disable these projects when pinot-fastdev is enabled.

Basically we usually do not need to compile these projects and what we don't want is to slow down the developer cycle just to shade a project that is not needed to develop a normal Pinot feature.